examples/Dockerfile

```diff
@@ -0,0 +1,4 @@
+FROM oak_docker:latest
+
+# This runs in the context of each example, so we just copy the bin
+COPY bin/config.bin .
```

Comment:
Which example does it copy?
Or it is being run sequentially for every example?
Is `bin/config.bin` rewritten every time a new example is built?

Reply:
Docker has a "context" in which is ran. You can think of it as the "parent folder". For each example image, I use the folder examples/EXAMPLE_NAME as the context, so bin/config.bin is the file for that e.g. examples/hello_world/bin/config.bin

Reply:
It does read a bit weird though. I wonder whether having an explicit `ARG config_file` which is passed as `docker build --build-arg=config_file=...` would be more readable.

Reply:
What does read a bit weird even mean? This is how the Dockerfiles work... maybe the comment should be different?

Reply:
The weirdness I was referring to is that usually a Docker image is built from a known location, but here you are expecting to move it around and be able to rebuild it in other folders.

Reply:
We are building an image of an example, so it kind make sense.. but I do see where you are coming from.